### PR TITLE
Update Skill Manager for nav,place to take target locations

### PR DESCRIPTION
**Pull request by KavitShah1998**
This PR is a replacement for https://github.com/facebookresearch/spot-sim2real/pull/74

### CODE REVIEW

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 1]
```diff
@@ -22,6 +23,49 @@
 
 
 class SpotSkillManager:
+    """
+    Class to manage the skills for Spot
```
Comment (zephirefaith) [inline spot_rl_experiments/spot_rl/envs/skill_manager.py:27]: Maybe change to: "Interface class to invoke skills for Spot, also exposes..."

"manage the skills" reads pretty vague.

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 2]
```diff
@@ -87,11 +132,14 @@ def nav(self, nav_target: str = None) -> Tuple[bool, str]:
             bool: True if navigation was successful, False otherwise
             str: Message indicating the status of the navigation
         """
+        print(f"Received nav target request for - {nav_target}")
```
Comment (zephirefaith) [inline spot_rl_experiments/spot_rl/envs/skill_manager.py:135]: Can we add a verbose flag here and put print statements under that?

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 3]
```diff
@@ -76,7 +120,8 @@ def reset(self):
         # Reset the policies and environments via the controllers
         raise NotImplementedError
 
-    def nav(self, nav_target: str = None) -> Tuple[bool, str]:
+    @multimethod  # type: ignore
+    def nav(self, nav_target: str = None) -> Tuple[bool, str]:  # type: ignore
         """
         Perform the nav action on the specified navigation target
```
Comment (zephirefaith) [inline spot_rl_experiments/spot_rl/envs/skill_manager.py:126]: Add "specified as a known string" at the end

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 4]
```diff
@@ -100,20 +148,40 @@ def nav(self, nav_target: str = None) -> Tuple[bool, str]:
...
 
-        print(f"Navigating to {nav_target}")
+        nav_x, nav_y, nav_theta = nav_target_tuple
+        status, message = self.nav(nav_x, nav_y, nav_theta)
+        return status, message
+
+    @multimethod  # type: ignore
+    def nav(self, x: float, y: float, theta=float) -> Tuple[bool, str]:  # noqa
+        """
+        Perform the nav action on the specified navigation target (location)
```
Comment (zephirefaith) [inline spot_rl_experiments/spot_rl/envs/skill_manager.py:158]: Explicit definition of what location means would be good here, i.e. location = x,y,theta

Comment (zephirefaith) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py:158]: I see you do it in args, maybe then just change to "specified navigation target represented as a metric location"

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 2 above)
Comment (zephirefaith) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py:135]: Or maybe add a global "verbose" flag to the SkillManager constructor, then every fn can print or not print based on requirement.

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 5]
```diff
@@ -100,20 +148,40 @@ def nav(self, nav_target: str = None) -> Tuple[bool, str]:
         else:
             return False, "No nav target specified, skipping nav"
 
-        print(f"Navigating to {nav_target}")
+        nav_x, nav_y, nav_theta = nav_target_tuple
+        status, message = self.nav(nav_x, nav_y, nav_theta)
+        return status, message
+
+    @multimethod  # type: ignore
```
Comment (zephirefaith) [inline spot_rl_experiments/spot_rl/envs/skill_manager.py:155]: nice!

**Review by zephirefaith — APPROVED** [review]
Documentation nits, but rest looks good. This is a LGTM from me but add the documentation clarification + verbose flag before merge, please.

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 1 above)
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: Addressed

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 2 above)
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: Added a new method `_conditional_print(msg)` which prints based on the verbose flag

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 4 above)
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: Updated

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 3 above)
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: Addressed

**Review by akshararai — APPROVED** [review]
looks good overall, some minor comments

**Review by jimmytyyang — APPROVED** [review]
LGTM! Please address the comments from the team!

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 6]
```diff
@@ -36,20 +81,22 @@ def __init__(self):
         self.waypoints_yaml_dict = get_waypoint_yaml()
 
     def __del__(self):
-        self.spot.shutdown(should_dock=True)
+        pass
```
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py:84]: We have a function call to "dock" which handles shutdown inside it.

This is the order of things
```
def function() {
    # Init skill manager
    spotskillmanager = SpotSkillManager()

    # Do some stuff
    status, msg = spotskillmanager.place(x, y, z)

    # Dock & then shutdown
    status, msg = spotskillmanager.dock()
}

```

When you call shutdown, robot actually powers off .... due to this, the "spot" object internally becomes `None`.

If we also keep `shutdown()` in the destructor, this `shutdown()` will try to reference `spot` object that become None as it as already shutdown.

That was really messy so we don't want to do shutdown inside the destructor.

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 7]
```diff
@@ -36,20 +81,22 @@ def __init__(self):
...
+        self.verbose = True
 
         # Create Spot object
         self.spot = Spot("RealSeqEnv")
 
         # Acquire spot's lease
         self.lease = self.spot.get_lease(hijack=True)
         if not self.lease:
-            print("Failed to get lease for Spot. Exiting!")
+            self._conditional_print("Failed to get lease for Spot. Exiting!")
```
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py:99]: Moved this function to utils.py

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 8]
```diff
@@ -122,84 +208,127 @@ def nav(self, nav_target: str = None) -> Tuple[bool, str]:
...
+                message = f"Failed - place target {place_target} not found - use the exact name"
+                self._conditional_print(message)
+                return False, message
         else:
-            return False, "No place target specified, skipping place"
+            message = "No place target specified, skipping place"
+            self._conditional_print(message)
+            return False, message
+
+        place_x, place_y, place_z = np.round(
```
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py:286]: Good catch, I think this was unintentional.

Thank you for catching this bug @akshararai

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: addressed here - 002d92fb57d51eac93c3abdf747aa131e248cd93

File: spot_rl_experiments/spot_rl/envs/skill_manager.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: addressed here - 002d92fb57d51eac93c3abdf747aa131e248cd93

File: spot_rl_experiments/spot_rl/envs/skill_manager.py [hunk 9]
```diff
@@ -72,48 +119,87 @@ def __initiate_controllers(self):
...
+                return False, message
         else:
-            return False, "No nav target specified, skipping nav"
+            msg = "No nav target specified, skipping nav"
+            return False, msg
 
-        print(f"Navigating to {nav_target}")
+        nav_x, nav_y, nav_theta = nav_target_tuple
+        status, message = self.nav(nav_x, nav_y, nav_theta)
+        self._conditional_print(message)
```
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply spot_rl_experiments/spot_rl/envs/skill_manager.py]: addressed here - 002d92fb57d51eac93c3abdf747aa131e248cd93

**Comment by KavitShah1998** [comment]
Merging the PR as approved by all

**Pull request merged**